controllers/productcontroller.go

```
@@ -0,0 +1,74 @@
+	"encoding/json"
+	"golang-crud-rest-api/database"
+	"golang-crud-rest-api/entities"
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+func CreateProduct(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var product entities.Product
+	json.NewDecoder(r.Body).Decode(&product)
```

golangci-lint (1.64.8)

Error return value of `(*encoding/json.Decoder).Decode` is not checked

(errcheck)